src/Evolution/Systems/Ccz4/Ricci.cpp

```diff
@@ -0,0 +1,141 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/Ccz4/Ricci.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace Ccz4 {
+template <size_t Dim, typename Frame, typename DataType>
+void ricci(
+    const gsl::not_null<tnsr::ii<DataType, Dim, Frame>*> result,
+    const tnsr::Ijj<DataType, Dim, Frame>& christoffel_second_kind,
+    const tnsr::iJkk<DataType, Dim, Frame>& d_conformal_christoffel_second_kind,
+    const tnsr::ii<DataType, Dim, Frame>& conformal_spatial_metric,
+    const tnsr::II<DataType, Dim, Frame>& inverse_conformal_spatial_metric,
+    const tnsr::ijj<DataType, Dim, Frame>& field_d,
+    const tnsr::iJJ<DataType, Dim, Frame>& field_d_up,
+    const tnsr::i<DataType, Dim, Frame>& field_p,
+    const tnsr::ij<DataType, Dim, Frame>& d_field_p) noexcept {
+  destructive_resize_components(result,
+                                get_size(get<0, 0>(conformal_spatial_metric)));
+  for (auto& component : (*result)) {
+    component = 0.0;
+  }
+
+  for (size_t i = 0; i < Dim; i++) {
+    for (size_t j = i; j < Dim; j++) {
+      for (size_t m = 0; m < Dim; m++) {
+        result->get(i, j) +=
```

Comment:
It would help if you wrote out the equation you're implementing in the documentation. It's much harder to check that the code is correct without knowing what you coded :). Adding comments like `// Add first term of d_m Gamma^m_{ij} and -d_j Gamma^m_{im}` also helps guide the reader with what you've actually implemented